packages/core/src/components/DropdownNew/hooks/useDropdownCombobox.ts

```diff
@@ -1,12 +1,14 @@
 import { useCallback, useMemo } from "react";
 import { useCombobox } from "downshift";
 import useDropdownFiltering from "./useDropdownFiltering";
-import { BaseListItemProps } from "../../BaseListItem";
-import { ListGroup } from "../../BaseList";
+import { BaseListItemData } from "../../BaseListItem";
 import { DropdownGroupOption } from "../Dropdown.types";
 
-function useDropdownCombobox<T extends BaseListItemProps>(
+function useDropdownCombobox<T extends BaseListItemData<Record<string, unknown>>>(
```

Comment:
same for the extends. It also shouldn't have any default to force us to pass the T to it, so I'm not sure what should this accept

Reply:
let's talk about it